common/libs/VkCodecUtils/ProgramConfig.h

```diff
@@ -66,88 +76,215 @@ struct ProgramConfig {
+    using ProgramArgs = std::vector<ArgSpec>;
+    static bool showHelp(const char ** argv, const ProgramArgs &spec) { 
+        std::cout << argv[0] << std::endl;
+        for ( auto& flag : spec ) {
+            std::stringstream ss;
+            if (flag.flag) {
+                ss << flag.flag << (flag.short_flag ? ", " : "");
+            }
+            if (flag.short_flag) {
+                ss << flag.short_flag;
+            }
+            // Print flags column 30 chars wide, left justified
+            std::cout << " " << std::left << std::setw(30) << ss.str();
+            // Print help if available, left justified
+            if (flag.help) {
+                std::cout << flag.help;
+            }
+            std::cout << std::endl;
+        }
+        return true;
+    };
+
+    void ParseArgs(int argc, const char *argv[]) {
+        ProgramArgs spec = {
+            {"--help", nullptr, 0, "Show this help",
+                [argv](const char **, const ProgramArgs &a) {
+                    int rtn = showHelp(argv, a);
+                    exit(EXIT_SUCCESS);
+                    return rtn;
+                }},
+            {"--enableStrDemux", nullptr, 0, "Enable stream demuxing",
+                [this](const char **, const ProgramArgs &a) {
+                    enableStreamDemuxing = true;
+                    return true;
+                }},
+            {"--disableStrDemux", nullptr, 0, "Disable stream demuxing",
+                [this](const char **, const ProgramArgs &a) {
+                    enableStreamDemuxing = true;
+                    return true;
+                }},
+            {"--codec", nullptr, 1, "Codec to decode",
+                [this](const char **args, const ProgramArgs &a) {
+                    if ((strcmp(args[0], "hevc") == 0) ||
+                        (strcmp(args[0], "h265") == 0)) {
+                        forceParserType = VK_VIDEO_CODEC_OPERATION_DECODE_H265_BIT_KHR;
+                        return true;
+                    } else if ((strcmp(args[0], "avc") == 0) ||
+                        (strcmp(args[0], "h264") == 0)) {
+                        forceParserType = VK_VIDEO_CODEC_OPERATION_DECODE_H264_BIT_KHR;
+                        return true;
+                    } else {
+                        std::cerr << "Invalid codec \"" << args[0] << "\"" << std::endl;
+                        return false;
+                    }
+                }},
+            {"--disableVsync", "-b", 0, "Disable vsync",
+                [this](const char **args, const ProgramArgs &a) {
+                    vsync = false;
+                    return true;
+                }},
+            {"--initialWidth", "-w", 1, "Initial width of the video",
+                [this](const char **args, const ProgramArgs &a) {
+                    initialWidth = std::atoi(args[0]);
+                    return true;
+                }},
+            {"--initialHeight", "-h", 1, "Initial height of the video",
```

Comment:
before, I changed `-h` from `--initialHeight` to `-h` for `--help` since i was always trying to use `vk-decode-app -h` to list the options. I changed this back following Tymurs feedback.